### ISSUE TITLE
install on cloud drive

### ISSUE THREAD

**Issue by N6REJ** [crazy-max/neard#258]
I'm sorry if I'm being dense but I don't see a place to ask this question anywhere else and I don't see the how to in the docs...
I want to install neard on my external drive which is synced to the cloud, my laptop will access these same files via the cloud also.. so for example..
desktop :   d:\amazon-cloud\neard\
laptop:   c:\amazon-cloud\neard\

What do I need to do to make this happen?

**Comment by crazy-max** [comment]
Hi @N6REJ,

Good question!

Synchronizing data between two computers with Neard via the cloud is possible, but only one instance of Neard must be executed, otherwise data may be corrupted because some files are locked. For example, MySQL that has its data locked when an instance is run and other services may also be impacted.

What it would be wise to develop on Neard in the future is to detect if another instance is already launched and at that time do not allow launching to avoid data corruption.

Does that answer your question?

**Comment by N6REJ** [comment]
that would be excellent, but what are the mechanics to doing so?  Clearly you need to install piece on both boxes.

**Comment by N6REJ** [comment]
just remembered that there is a method for db's that are access from multiple terminals ( think of banks ) but I can't remember what that required.. just a vague memory.

**Comment by crazy-max** [comment]
I will take an example with **computer1** and **computer2**.

### On computer1

In the directory where the folders are synchronized your files on the cloud (we will say `C:\cloud`), extract Neard. This will give us: `C:\cloud\neard`.
Do not run Neard and wait for the synchronization to complete.

### On computer2

Have the same path for synchronization : `C:\cloud\neard`
Wait for data from the cloud that has been synchronized from the computer1.

### Usage

Run Neard on computer1 and work on one of your projects. Once you have finished working, close Neard via the notification area icon.

If after you want to work on the computer2, make sure that Neard has been shut down on computer1 and the data has been synchronized. Once done, you can start Neard on computer2.

### To summarize

Be careful that there is only one instance of Neard launched at a time and also that the files are synchronized before running on another computer

**Comment by N6REJ** [comment]
that's exactly what I perceive happening... but there is a binary that needs to be installed on both isn't there?

**Comment by crazy-max** [comment]
> just remembered that there is a method for db's that are access from multiple terminals ( think of banks ) but I can't remember what that required.. just a vague memory.

It would be possible if we have a database on an other computer. For example **computer3** with IP address 192.168.0.3 will be dedicated for external access to MySQL.
Then your projects will have to use the database from computer3 (192.168.0.3) and not localhost as usual.

**Comment by crazy-max** [comment]
> that's exactly what I perceive happening... but there is a binary that needs to be installed on both isn't there?

Only [Neard Prerequisites Package](https://github.com/crazy-max/neard-prerequisites/releases/latest) have to installed on each computer that's it.

**Comment by N6REJ** [comment]
ok, I'll go try more thoroughly before I ask more questions.. ty.

**Comment by N6REJ** [comment]
ok @crazy-max I've confirmed the problem.. I don't understand it, but I know WHAT is happening...
the users in mysql don't match on each machine even though the database is exactly the same file(s).  
This is the laptop... which is the secondary machine.. i.e. the one working from google drive.  
![image](https://cloud.githubusercontent.com/assets/1850089/26707050/996cfb6c-4707-11e7-8f08-616738676b56.png)
This is the laptop... which is the secondary machine.. i.e. the one working from google drive.

And this is the desktop where wamp is installed via windows..  
![image](https://cloud.githubusercontent.com/assets/1850089/26707113/f84bf8a4-4707-11e7-94a6-fb8b3f19a863.png)

Very very very strange.

**Comment by N6REJ** [comment]
@crazy-max I think I came up with a better solution.  We both know the dangers of using single db from multiple pcs.  So what if we used https://portableapps.com/apps/development/portableapps.com_installer instead?  xampp does this so I would think neard could also.  Then there is no need for trickery.  Just put portable on both machines and the rest is done automatically.

**Comment by crazy-max** [comment]
@N6REJ A Portableapps can be done but the result will be the same i think. Have you tried with Xampp and a single db ?

**Comment by N6REJ** [comment]
thats on the schedule for this weekend.

**Comment by crazy-max** [comment]
I close this issue for the momenbt waiting your answer
